internal-dns/src/dropshot_server.rs

```diff
@@ -43,31 +44,31 @@ async fn dns_records_get(
 }
 
 #[endpoint(
-    method = PUT,
-    path = "/set-records",
+    method = POST,
```

Comment:
Re: PUT vs POST, this method *could* be idempotent? Each record is inserted into the DNS db as follows:

https://github.com/oxidecomputer/omicron/blob/6ca28366111e3a82b806a9456a9156d45cc893cb/internal-dns/src/dns_data.rs#L308-L320

That insert is doing an "insert, or replace if the key already exists".

I don't feel that strongly, but just a heads up, that seems kinda at-odds with the POST request, where new records won't actually be created on the next call.

Reply:
Ah, yes, I see we are having a "I don't have a strong opinion" standoff.

If you think it's more consistent with `POST`, I say go for it.